src/ring_theory/fractional_ideal.lean

```diff
@@ -1078,6 +1078,32 @@ begin
   rw [algebra.smul_def, mul_one]
 end
 
+lemma span_singleton_eq_span_singleton [no_zero_smul_divisors R P] {x y : P} :
+  span_singleton S x = span_singleton S y ↔ ∃ z : Rˣ, z • x = y :=
```

Comment:
Is this true for submodule too? If so, does this result follow trivially from that one?